src/utils.cpp

```diff
@@ -477,7 +477,7 @@ int modify_target_rbuf_raw(struct char_data *ch, char *rbuf, size_t rbuf_len, in
 
   // If you're astrally perceiving, you don't take additional vision penalties, and shouldn't have any coming in here.
   if (SEES_ASTRAL(ch)) {
-    if (!skill_is_magic && IS_PERCEIVING(ch)) {
+    if (!skill_is_magic && IS_PERCEIVING(ch) && !IS_GHOUL(ch)) {
```

Comment:
The effect of this is that ghouls can astrally perceive (bypassing vision penalties) without the +2 TN penalty, is that intended? That will make them very powerful as awakened since they can leave perceive on 24/7.

Reply:
Intended, actually. Ghouls are Dual Natured but receive no penalties associated with astral perception when doing physical tasks. It's one of the biggest boons they have (but also makes them susceptible to astral entities 24/7 since they can't turn it off). I was actually hoping to get your help on this since technically Mundane Ghouls should also have this, as well as IS_DRAGON types, so it needs some tweaking